cmd/pmm-configurator/instance.go

```diff
@@ -17,12 +17,11 @@ func checkInstanceHandler(w http.ResponseWriter, req *http.Request) {
 		return
 	}
 
-	result, err := checkInstance(passedInstance.ID)
-	if result == "success" {
-		returnSuccess(w)
-	} else {
+	if result, err := checkInstance(passedInstance.ID); result != "success" {
```

Comment:
string `success` has 10 occurrences, make it a constant

Reply:
Almost correct golangci ;)
I would say use correctly `err` where `err == nil` means success. You won't need "success" word.

Reply:
not really, it is needed to pass human-readable error text and error itself, it is possible to create own error type which will include both human-readable text and nested error.
I think it is not needed to hide return structure in own classes because it is possible to forget to pass human-readable error.

Reply:
Just use `fmt.Errorf("human readable error: %s", err)` - that's the practice. Yes, you can also use even more complex structures https://blog.golang.org/error-handling-and-go